src/containers/LogInContainer.js

```diff
@@ -33,7 +33,16 @@ const shouldComponentUpdate = (props, prevProps) => {
   const { match: pMatch, ...prest } = prevProps
   const { match, ...rest } = props
   if (localStorage.getItem('authToken')) {
-    props.history.push('/profiili')
+    API.getUser(
+      localStorage.getItem('userId'),
+      localStorage.getItem('authToken')
+    ).then(loggedUser => {
+      if (loggedUser.profileReady) {
```

Comment:
I couldn't get this to work at first but then realised that it must be because profileReady is not being set after profile creation flow. Apart from that looks fine to me :)